### ISSUE TITLE
Tracking and counting individual objects

### ISSUE THREAD

**Issue by samuel-cox** [AlexeyAB/darknet#1849]
I'm aiming to document a live stream feed from my webcam, using Darknet to detect objects then output using -ext_output. All works great.

However, i'm looking to count the amount of times an object is in the scene. I.E - if i'm showing a busy road, I want to know how many times a bus, a car, a person etc enters the image.

As it stands, -ext_output constantly streams what's in each frame, not a count. Is there any functionality that enables such logging?

Thanks in advance

**Comment by PeterQuinn925** [comment]
No, but you can call it from another program that does the counting.

**Comment by samuel-cox** [comment]
@PeterQuinn925 yes, that's my fall-back. Only issue with this method is that one bus might be in frame for 5 seconds (running at 25fps thats 125 counts of one bus), where I need to print out as one until it's existed the frame.

So my question then falls to... is there any way of using -ext_output to only trigger once per object detected, until that objects last left the frame?

**Comment by PeterQuinn925** [comment]
How is it supposed to know it's the same bus? Each frame is detected independently.  You'll need your program too figure out if it's the same object or not and to count it as a new one or not.  There's probably some OpenCV functions that can help. Or, something as simple as if consecutive frames have the same class of object, it's probably the same object might work well enough.
